src/main/java/pl/javastart/exercise/streams/FilteringMachine.java

```diff
@@ -1,23 +1,35 @@
 package pl.javastart.exercise.streams;
 
 import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class FilteringMachine {
 
     public List<Integer> filterOutNotEvenNumbers(List<Integer> numberList) {
-        return null;
+        return numberList.stream()
+                .filter(number -> number % 2 == 0)
+                .collect(Collectors.toList());
     }
 
     public List<Integer> filterOutLowerNumbersThan20(List<Integer> numberList) {
-        return null;
+        return numberList.stream()
+                .filter(number -> number >= 20)
+                .collect(Collectors.toList());
     }
 
     public List<Book> convertToBooks(List<String> titles) {
-        return null;
+        return titles.stream()
+                .map(Book::new)
+                .collect(Collectors.toList());
     }
 
     // Metoda powinna przekształcić tytuły na książki i zwrócić tylko te które rozpoczynają się od słowa "Gra"
     public List<Book> convertToBooksAndReturnOnlyStartingWithGra(List<String> titles) {
-        return null;
+        return titles.stream()
+                .map(Book::new)
+                .filter(book -> book.getTitle().startsWith("Gra"))
```

Comment:
Tutaj lepiej by było najpierwsz przefiltrować tytuły a potem stworzyć książki za pomocą metody map. Jest to istotne bo nie zamienisz od razu wszystkich książek na obiekty, tylko te które rzeczywiście tymi książkami powinny być. Taka oszczędność pamięci troche :)

Reply:
Poprawiłem. Nie przeanalizowałem dokładnie zadania i zrobiłem w kolejności, w jakiej było w nazwie metody. Będę pamiętał, żeby analizować kod pod względem optymalizacji.